packages/epetra/test/IntMultiVectorDistributed/cxx_main.cpp

```diff
@@ -0,0 +1,183 @@
+//
+// 2. Redistributions in binary form must reproduce the above copyright
+// notice, this list of conditions and the following disclaimer in the
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY SANDIA CORPORATION "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL SANDIA CORPORATION OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Michael A. Heroux (maherou@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+*/
+
+#include "Epetra_ConfigDefs.h"
+#include "Epetra_IntMultiVector.h"
+#include "Epetra_BlockMap.h"
+#include "Epetra_CombineMode.h"
+#include "Epetra_Import.h"
+#ifdef EPETRA_MPI
+#include "Epetra_MpiComm.h"
+#include <mpi.h>
+#else
+#include "Epetra_SerialComm.h"
+#endif
+#include "../epetra_test_err.h"
+#include "Epetra_Version.h"
+
+int main(int argc, char *argv[]) {
+  int ierr = 0;
+  int global_ierr = 0;
+  int numErr = 0;
+
+#ifdef EPETRA_MPI
+
+  // Initialize MPI
+
+  MPI_Init(&argc,&argv);
+  int rank; // My process ID
+
+  MPI_Comm_rank(MPI_COMM_WORLD, &rank);
+  Epetra_MpiComm Comm( MPI_COMM_WORLD );
+
+#else
+
+  int rank = 0;
+  Epetra_SerialComm Comm;
+
+#endif
+
+  bool verbose = false;
+
+  // Check if we should print results to standard out
+  if(argc > 1) {
+    if(argv[1][0]=='-' && argv[1][1]=='v') {
+      verbose = true;
+    }
+  }
+
+  Comm.SetTracebackMode(0); // This should shut down any error traceback reporting
+  int MyPID = Comm.MyPID();
+  int NumProc = Comm.NumProc();
+
+  if(verbose && MyPID==0)
+    cout << Epetra_Version() << endl << endl;
+
+  if(verbose) cout << "Processor " << MyPID << " of " << NumProc << " is alive." << endl;
+
+  // Redefine verbose to only print on PE 0
+  if(verbose && rank != 0)
+		verbose = false;
+
+  int NumMyEquations = 6;
+  int NumGlobalEquations = NumMyEquations*NumProc+EPETRA_MIN(NumProc,3);
+  if(MyPID < 3)
+    NumMyEquations++;
+
+  // Construct a Map that puts approximately the same Number of equations on each processor
+
+  Epetra_BlockMap& Map = *new Epetra_BlockMap(NumGlobalEquations, NumMyEquations, 1, 0, Comm);
```

Comment:
1. Don't do this.  Please instead assign to a pointer.
2. This leaks memory, since you never free the object.

Reply:
@mhoemmen I am actually tempted to simply change this to:
```
Epetra_BlockMap Map(NumGlobalEquations, NumMyEquations, 1, 0, Comm);
```
which would work just fine and run out off scope, thus avoiding the memory leak.
Is there a good reason to use a pointer to the object instead of the object directly?

Reply:
@lucbv I like this use of values instead of pointers.  I don't see a reason to use pointers here -- if you only ever keep the reference, then the value should also do fine.